common-npm-packages/webdeployment-common/fileTransformationsUtility.ts

```diff
@@ -138,34 +141,56 @@ export function enhancedFileTransformations(isFolderBasedDeployment: boolean, xm
             }
             else {
                 tl.error(tl.loc('FailedToApplySpecialTransformationReason1'));
-            }          
+            }
         }
     }
 
     let isSubstitutionApplied: boolean = true;
-    if(xmlTargetFiles.length > 0) 
-    {     
-        xmlTargetFiles.forEach(function(fileName) { 
+    if(xmlTargetFiles.length > 0)
+    {
+        xmlTargetFiles.forEach(function(fileName) {
             isSubstitutionApplied = xmlSubstitutionUtility.substituteAppSettingsVariables(folderPath, isFolderBasedDeployment, fileName) || isSubstitutionApplied;
         });
-        
+
         if(isSubstitutionApplied) {
-            console.log(tl.loc('XMLvariablesubstitutionappliedsuccessfully')); 
-        } 
+            console.log(tl.loc('XMLvariablesubstitutionappliedsuccessfully'));
+        }
         else {
             tl.error(tl.loc('FailedToApplyXMLvariablesubstitutionReason1'));
         }
     }
 
     isSubstitutionApplied = true;
-    if(jsonTargetFiles.length > 0) 
+    if(jsonTargetFiles.length > 0)
     {
         isSubstitutionApplied = jsonSubstitutionUtility.jsonVariableSubstitution(folderPath, jsonTargetFiles, true);
         if(isSubstitutionApplied) {
-            console.log(tl.loc('JSONvariablesubstitutionappliedsuccessfully')); 
-        } 
+            console.log(tl.loc('JSONvariablesubstitutionappliedsuccessfully'));
+        }
         else {
             tl.error(tl.loc('FailedToApplyJSONvariablesubstitutionReason1'));
         }
     }
 }
+
+export async function applyTransformations(webPackage: string, parameters: string, packageType: PackageType): Promise<string> {
+    tl.debug("WebConfigParameters is "+ parameters);
+    if (parameters) {
+        var isFolderBasedDeployment: boolean = tl.stats(webPackage).isDirectory();
+        var folderPath = await deployUtility.generateTemporaryFolderForDeployment(isFolderBasedDeployment, webPackage, packageType);
+        if (parameters) {
+            tl.debug('parsing web.config parameters');
+            var webConfigParameters = parse(parameters);
+            const rootDirectoryPath: string = "D:\\home\\site\\wwwroot";
```

Comment:
Are we sure this path string is always same?

Reply:
It is like this in the original library from which it is migrated - https://github.com/microsoft/azure-pipelines-tasks-common-packages/blob/main/common-npm-packages/azurermdeploycommon/operations/FileTransformsUtility.ts#L9